File: src/modules/auth/controllers/auth.controller.ts
import { Response } from 'express';
import { IRequest } from '../../../shared/domain/interfaces/request.interface';
import { AuthLoginDto } from '../domain/dto/auth-login.dto';
import { authLoginService } from '../services/auth-login.service';
import { AuthRecoverPasswordDto } from '../domain/dto/auth-recover-password.dto';
import { authRecoverPasswordService } from '../services/auth-recover-password.service';
import { AuthResetPasswordDto } from '../domain/dto/auth-reset-password.dto';
import { authResetPasswordService } from '../services/auth-reset-password.service';
import { AuthRefreshDto } from '../domain/dto/auth-refresh.dto';
import { authRefreshService } from '../services/auth-refresh.service';
import { AuthRegisterDto } from '../domain/dto/auth-register.dto';
import { authRegisterService } from '../services/auth-register.service';
import { generateJsonError } from '../../../shared/utils/error';

class AuthController {
  async login(request: IRequest, response: Response) {
    const dto: AuthLoginDto = request.body;
    try {
      response.status(200).json(await authLoginService.execute(dto));
    } catch (error) {
      generateJsonError(error, response);
    }
  }

  async refresh(request: IRequest, response: Response) {
    const dto: AuthRefreshDto = request.body;
    try {
      response
        .status(200)
        .json(await authRefreshService.execute(dto.refreshToken));
    } catch (error) {
      generateJsonError(error, response);
    }
  }

  async register(request: IRequest, response: Response) {
    const dto: AuthRegisterDto = request.body;
    try {
      response.status(201).json(await authRegisterService.execute(dto));
    } catch (error) {
      generateJsonError(error, response);
    }
  }

  async recoverPassword(request: IRequest, response: Response) {
    const dto: AuthRecoverPasswordDto = request.body;
    try {
      response.status(200).json(await authRecoverPasswordService.execute(dto));
    } catch (error) {
      generateJsonError(error, response);
    }
  }

  async resetPassword(request: IRequest, response: Response) {
    const dto: AuthResetPasswordDto = request.body;
    try {
      response.status(200).json(await authResetPasswordService.execute(dto));
    } catch (error) {
      generateJsonError(error, response);
    }
  }
}

export const authController = new AuthController();

File: src/modules/authentication/controllers/auth_controller.py
from ninja import Router
from src.modules.authentication.domain.dto.auth_login_dto import AuthLoginDTO
from src.modules.authentication.domain.dto.auth_login_response_dto import (
    AuthLoginResponseDTO,
)
from src.modules.authentication.services.auth_login_service import AuthLoginService
from src.modules.authentication.domain.dto.auth_refresh_dto import AuthRefreshDTO
from src.modules.authentication.domain.dto.auth_refresh_response_dto import (
    AuthRefreshResponseDTO,
)
from src.modules.authentication.services.auth_refresh_service import AuthRefreshService
from src.modules.authentication.domain.dto.auth_recover_password_dto import (
    AuthRecoverPasswordDTO,
)
from src.modules.authentication.domain.dto.auth_recover_password_response_dto import (
    AuthRecoverPasswordResponseDTO,
)
from src.modules.authentication.services.auth_recover_password_service import (
    AuthRecoverPasswordService,
)
from src.modules.authentication.services.auth_reset_password_service import (
    AuthResetPasswordService,
)
from src.modules.authentication.domain.dto.auth_reset_password_dto import (
    AuthResetPasswordDTO,
)
from src.modules.user.domain.schemas.user_schema import UserSchema
from src.modules.authentication.services.auth_register_service import (
    AuthRegisterService,
)
from src.modules.authentication.domain.dto.auth_register_dto import (
    AuthRegisterDTO,
)
from src.shared.decorators.public_decorator import public

router = Router(tags=["Auth"])


@router.post("/login", response=AuthLoginResponseDTO, auth=None)
@public
def login(request, dto: AuthLoginDTO):
    return AuthLoginService.execute(dto)


@router.post("/refresh", response=AuthRefreshResponseDTO, auth=None)
@public
def refresh(request, dto: AuthRefreshDTO):
    return AuthRefreshService.execute(dto)


@router.post("/register", response=UserSchema, auth=None)
@public
def register(request, dto: AuthRegisterDTO):
    return AuthRegisterService.execute(dto)


@router.post(
    "/recover-password",
    response=AuthRecoverPasswordResponseDTO,
    auth=None,
)
@public
def recover_password(request, dto: AuthRecoverPasswordDTO):
    return AuthRecoverPasswordService.execute(dto)


@router.patch("/reset-password", response=UserSchema, auth=None)
@public
def reset_password(request, dto: AuthResetPasswordDTO):
    return AuthResetPasswordService.execute(dto)
